src/Symfony/Component/Intl/Locale.php

```diff
@@ -67,6 +67,21 @@ public static function getDefaultFallback()
      */
     public static function getFallback($locale)
     {
+        if (function_exists('locale_parse')) {
+            $localesSubTags = locale_parse($locale);
+            if (1 === count($localesSubTags)) {
+                if (self::$defaultFallback === $localesSubTags['language']) {
+                    return 'root';
+                }
+
+                return self::$defaultFallback;
+            }
+
+            array_pop($localesSubTags);
+
+            return locale_compose($localesSubTags);
+        }
+
         if (false === $pos = strrpos($locale, '_')) {
```

Comment:
Can we also make the fallback work with `-`?

Reply:
Done